doc/cmake.md

```diff
@@ -0,0 +1,33 @@
+# Building a new data model library with CMake
+
+PODIO exposes multiple CMake macros to enable the building of data model libraries.
+For a package developer tutorial on CMake, please have a look [here])https://hsf-training.github.io/hsf-training-cmake-webpage/). PODIO follows the same conventions used therein.
+
+## Generate the C++ code
+After a proper `find_package(PODIO)`, the C++ code of the new data model `NewDM` can be created and compiled via
+
+```
+# generate the c++ code from the yaml definition
+PODIO_GENERATE_DATAMODEL(newdm newdm.yaml headers sources IO_BACKEND_HANDLERS)
+
+# compile the core data model shared library (no I/O)
+PODIO_ADD_DATAMODEL_CORE_LIB(newdm "${headers}" "${sources}")
+```
+
+However, this does not create any I/O components yet. For this, one needs to add the `IO_BACKEND_HANDLERS` parameter and then generate and compile the corresponding backends, like shown below for the ROOT and SIO backends:
```

Comment:
Actually by default we always build the `ROOT` backend (even if no `IO_BACKEND_HANDLERS` are passed explictly).